plugins/outputs/mqtt/template_parser.go

```diff
@@ -0,0 +1,47 @@
+package mqtt
+
+import (
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+type TemplateTopic struct {
+	Hostname    string
+	metric      telegraf.Metric
+	topicPrefix string
+}
```

Comment:
How about letting the struct contain all static elements and only use a transient metric
```suggestion
type TemplateTopic struct {
	Hostname    string
	TopicPrefix string
	Template *template.Template
	metric      telegraf.Metric
}
```